napari/components/viewer_model.py

```diff
@@ -334,7 +338,10 @@ def _on_layers_change(self):
             ndim = len(ranges)
             self.dims.ndim = ndim
             for i, _range in enumerate(ranges):
+                old_range = self.dims.range[i]
                 self.dims.set_range(i, _range)
+                if old_range != _range:
+                    self.dims.set_point(i, self.rounded_division(*_range))
```

Comment:
instead of simple centering, there could be some more complicated code to recalculate to a new position base on old.